extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -900,128 +891,213 @@ private void checkPlayabilityStatus(final JsonObject youtubePlayerResponse,
                             Objects.requireNonNullElse(detailedErrorMessage, reason));
                 }
             }
+
+            if (reason.contains("age-restricted")) {
+                throw new AgeRestrictedContentException(
+                        "This age-restricted video cannot be watched anonymously");
+            }
         }
 
         throw new ContentNotAvailableException("Got error: \"" + reason + "\"");
     }
 
-    /**
-     * Fetch the Android Mobile API and assign the streaming data to the androidStreamingData JSON
-     * object.
-     */
-    private void fetchAndroidMobileJsonPlayer(@Nonnull final ContentCountry contentCountry,
-                                              @Nonnull final Localization localization,
-                                              @Nonnull final String videoId)
+    private void fetchHtml5Client(@Nonnull final Localization localization,
+                                  @Nonnull final ContentCountry contentCountry,
+                                  @Nonnull final String videoId,
+                                  @Nullable final PoTokenProvider poTokenProviderInstance,
```

Comment:
Why don't we simply do a null-check rather than having 2 fields for this? (in all the fetch methods)

Reply:
It was to avoid recomputing the field nullability multiple times (also in a previous version when I was working on this, the `fetchHtml5Client` method content was directly in the `onFetchPage` method if I remember properly), but maybe this should be avoided.

Reply:
I'm pretty sure passing one more parameter around is slower than redoing the null check. Actually, I would guess that `if (o != null)` is as fast as `if (b)`, since both perform a memory load and then use the `zero` flag to determine whether to jump or not (assuming `null` is represented as 0).